src/RepoAuditor/CommandLineProcessor.py

```diff
@@ -130,6 +109,47 @@ def Create(
                     argument_separator.join(parts[1:])
                 )
 
+        return excluded_requirements
+
+    @classmethod
+    def Create(
+        cls,
+        get_dynamic_args_func: "CommandLineProcessor.GetDynamicArgsFunc",
+        modules: list[Module],
+        includes: list[str],
+        excludes: list[str],
+        warnings_as_error_module_names: set[str],
+        ignore_warnings_module_names: set[str],
+        *,
+        all_warnings_as_error: bool = False,
+        ignore_all_warnings: bool = False,
+        single_threaded: bool = False,
+        argument_separator: str = "-",
+    ) -> "CommandLineProcessor":
+        """Factor method to construct a CommandLineProcessor object."""
+        # Convert modules into a lookup map
+        module_map: dict[str, Module] = {}
+
+        for module in modules:
+            if argument_separator in module.name:
+                msg = f"The module name '{module.name}' contains '{argument_separator}', which should be used as an argument separator."
+                raise ValueError(msg)
+
+            prev_module = module_map.get(module.name, None)
+            if prev_module is not None:
+                msg = f"The module '{module.name}' has already been defined."
+                raise ValueError(msg)
+
+            module_map[module.name] = module
+
+        del modules
+
+        # Process includes
+        included_requirements = cls._process_includes(module_map, includes, argument_separator)
+        del includes

```

Comment:
I move two sub-tasks into separate class method. Not sure whether class methods are the most appropriate for that task, I'd appreciate some feedback on this specifically.

Reply:
Something similar was done in #95, so maybe we can hold off here and re-assess once #95 is merged.

Reply:
@mtanneau I'll merge in main and resolve the conflicts.